    public java.lang.String f() throws java.io.IOException
    {
        java.lang.Throwable $r5;
        byte[] $r2;
        java.io.ByteArrayOutputStream $r6;
        i.k.c.a.c.n r0;
        java.nio.charset.Charset $r3;
        int $i0;
        java.lang.String $r4;
        java.io.InputStream $r1;

        r0 := @this: i.k.c.a.c.n;

        $r1 = virtualinvoke r0.<i.k.c.a.c.n: java.io.InputStream b()>();

        if $r1 != null goto label1;

        return "";

     label1:
        $r6 = new java.io.ByteArrayOutputStream;

        specialinvoke $r6.<java.io.ByteArrayOutputStream: void <init>()>();

        $r2 = newarray (byte)[4096];

     label2:
        $i0 = virtualinvoke $r1.<java.io.InputStream: int read(byte[])>($r2);

     label3:
        if $i0 != -1 goto label4;

        virtualinvoke $r1.<java.io.InputStream: void close()>();

        $r3 = virtualinvoke r0.<i.k.c.a.c.n: java.nio.charset.Charset c()>();

        $r4 = virtualinvoke $r3.<java.nio.charset.Charset: java.lang.String name()>();

        $r4 = virtualinvoke $r6.<java.io.ByteArrayOutputStream: java.lang.String toString(java.lang.String)>($r4);

        return $r4;

     label4:
        virtualinvoke $r6.<java.io.OutputStream: void write(byte[],int,int)>($r2, 0, $i0);

     label5:
        goto label2;

     label6:
        $r5 := @caughtexception;

        virtualinvoke $r1.<java.io.InputStream: void close()>();

        throw $r5;

        catch java.lang.Throwable from label2 to label3 with label6;
        catch java.lang.Throwable from label4 to label5 with label6;
    }
